cp3pt0-deployment/migrate_tenant.sh

```diff
@@ -86,16 +85,20 @@ function main() {
     update_cscr "$OPERATOR_NS" "$SERVICES_NS" "$NS_LIST"
     accept_license "commonservice" "$OPERATOR_NS"  "common-service"
 
+    # Validate ibm-common-service-operator CatalogSource and CatalogSourceNamespaces
     # Update ibm-common-service-operator channel
     for ns in ${NS_LIST//,/ }; do
-        if [ $ENABLE_PRIVATE_CATALOG -eq 0 ]; then
-            check_cs_catalogsource
-            update_operator ibm-common-service-operator $ns $CHANNEL $SOURCE $SOURCE_NS $INSTALL_MODE
-        else
-            PRIVATE_NS=$ns
-            check_cs_catalogsource
-            update_operator ibm-common-service-operator $PRIVATE_NS $CHANNEL $SOURCE $PRIVATE_NS $INSTALL_MODE
+        local pm="ibm-common-service-operator"
+        local sub_name=$(${OC} get subscription.operators.coreos.com -n ${ns} -l operators.coreos.com/${pm}.${ns}='' --no-headers | awk '{print $1}')
+        if [ ! -z "$sub_name" ]; then
+            op_source=$SOURCE
+            op_source_ns=$SOURCE_NS
+            if [ $ENABLE_PRIVATE_CATALOG -eq 1 ]; then
+                op_source_ns=$ns
+            fi
         fi
+        validate_operator_catalogsource $pm $ns $op_source $op_source_ns $CHANNEL op_source op_source_ns
+        update_operator $pm $ns $CHANNEL $op_source $op_source_ns $INSTALL_MODE
```

Comment:
Could we move those two functions into above if condition? It will validate catalog and update operator when subscription exists.